spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -199,15 +199,15 @@ class CometSparkSessionExtensions
                 _,
                 _,
                 _)
-              if CometScanExec.isFileFormatSupported(fileFormat)
+              if COMET_EXEC_ENABLED.get()
+                && CometConf.isExperimentalNativeScan
```

Comment:
I moved this check earlier because it is much cheaper than checking if the schema is supported (especially if the schema is deeply nested)